### PR TITLE
Pbw 3181

### PR DESCRIPTION
**Pull request by chibzu**

### CODE REVIEW

File: js/components/controlBar.js [hunk 1]
```diff
@@ -229,25 +223,26 @@ var ControlBar = React.createClass({
       //create each volume tick separately
       var turnedOn = this.props.controller.state.volumeState.volume >= (i+1) / 10;
       var highlighted = this.state.mouseOverVolume;
-      var singleBarStyle = Utils.clone(InlineStyle.controlBarStyle.volumeBarStyle);
-      singleBarStyle.backgroundColor = (turnedOn ?
-        (highlighted?"rgba(67, 137, 255, 1)":"rgba(67, 137, 255, 0.6)") : (highlighted?"rgba(255, 255, 255, 1)":"rgba(255, 255, 255, 0.6)"));
-      //we store which value the tick correlates to via a data attribute on the element
-      volumeBars.push(<span data-volume={(i+1)/10} style={singleBarStyle}
+      var volumeClass = "volumeBar";
+      if (turnedOn) volumeClass += " on";
```
Comment (mistermendez) [inline js/components/controlBar.js]: consider using Classnames here -

var volumeClass = ClassNames({
      'volumeBar': true,
      'on': turnedOn
    });

File: js/components/controlBar.js [hunk 2]
```diff
@@ -229,25 +223,26 @@ var ControlBar = React.createClass({
...
+      if (highlighted) volumeClass += " highlighted";
```
Comment (mistermendez) [inline js/components/controlBar.js]: consider using CSS :hover for mouseover highlight

File: scss/components/_control-bar.scss [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+  background-color: rgba(255, 255, 255, 0.6);
+}
+
+.volumeBar.on {
+  background-color: rgba(67, 137, 255, 0.6);
+}
+
+.volumeBar.highlighted {
+  background-color: rgba(255, 255, 255, 1);
+}
```
Comment (mistermendez) [inline scss/components/_control-bar.scss]: consider -

.volumeBar:hover{
 +  background-color: rgba(255, 255, 255, 1);
 +}

File: tests/components/controlBar-test.js [hunk 4]
```diff
@@ -0,0 +1,656 @@
+jest.dontMock('../../js/components/controlBar')
+    .dontMock('../../js/components/utils')
+    .dontMock('../../js/constants/constants')
+    .dontMock('../../config/en.json')
```
Comment (mistermendez) [inline tests/components/controlBar-test.js]: not needed in this test

File: tests/components/controlBar-test.js [hunk 5]
```diff
@@ -0,0 +1,656 @@
...
+    .dontMock('../../config/es.json')
```
Comment (mistermendez) [inline tests/components/controlBar-test.js]: not needed in this test

File: tests/components/controlBar-test.js [hunk 6]
```diff
@@ -0,0 +1,656 @@
...
+    .dontMock('../../config/zh.json');
```
Comment (mistermendez) [inline tests/components/controlBar-test.js]: not needed in this test

File: js/components/controlBar.js [hunk 7]
```diff
@@ -196,13 +186,17 @@ var ControlBar = React.createClass({
 
   volumeHighlight:function() {
```
Comment (mistermendez) [inline js/components/controlBar.js]: could :hover replace this?

Comment (chibzu) [inline js/components/controlBar.js]: We still want to use the opacity/color settings from the skin config for the icon, so I think this part needs to stay in the JS.

Comment (mistermendez) [inline js/components/controlBar.js]: gotcha

**Comment by chibzu** [comment]
Closing this until the PR for the skin config change is ready.

**Pull request closed without merging**